src/main/CommandLine.cpp

```diff
@@ -957,11 +970,26 @@ int
 runNewDB(CommandLineArgs const& args)
 {
     CommandLine::ConfigOption configOption;
+    bool resetInMemoryState = false;
 
-    return runWithHelp(args, {configurationParser(configOption)}, [&] {
-        initializeDatabase(configOption.getConfig());
-        return 0;
-    });
+    auto resetParser = [](bool& resetInMemoryState) {
+        return clara::Opt{resetInMemoryState}["--reset-for-in-memory-mode"](
+            "Reset the special database used only for in-memory mode (see "
+            "--replay-in-memory flag");
```

Comment:
the `--replay-in-memory` option is itself marked as deprecated; should _reference_ `--in-memory` here.